pywhy_stats/kci.py

```diff
@@ -0,0 +1,196 @@
+from typing import Callable
+
+import numpy as np
+from numpy import shape, ndarray
+from numpy.linalg import eigvalsh
+from numpy.random import permutation
+from scipy import stats
+
+from pywhy_stats.custom_kernels import rbf_kernel
+from pywhy_stats.pvalue_result import PValueResult
+
+
+def ind(X: np.ndarray,
+        Y: np.ndarray,
+        kernel_X: Callable[[np.ndarray], np.ndarray] = rbf_kernel,
```

Comment:
Should we use `numpy.typing.ArrayLike` instead, which is their supported mypy typing extension? 

Ref: https://numpy.org/devdocs/reference/typing.html

Reply:
The issue I see with `ArrayLike` is that the IDE complains that it doesn't have properties such as `shape`, which means, it is not the same as an `ndarray`?

Reply:
That's weird. Shapes hasn't been implemented yet https://github.com/numpy/numpy/issues/16544, but maybe there's some issue w/ the IDE? 

But I think ArrayLike is more general because it will supposedly support any "array-API" (in the future). But yeah not super important until shapes implemented imo.